File: unity/Assets/Scripts/Multiplayer/Rooms/Updates/UpdatePlayersResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Rooms.Updates
{
    /// <summary>
    /// Response returned when updates are successfully sent to target players.
    /// </summary>
    [System.Serializable]
    public class UpdatePlayersResponse : ApiResponse<ERoomUpdatesError>
    {
        /// <summary>
        /// The number of updates that were sent.
        /// </summary>
        public int updates_sent;

        /// <summary>
        /// List of IDs for the sent updates.
        /// </summary>
        public List<string> update_ids = new();

        /// <summary>
        /// List of target player IDs for the updates.
        /// </summary>
        public List<int> target_players_ids = new();
    }
}

File: java/src/main/java/com/michitai/multiplayer/rooms/updates/UpdatePlayersResponse.java
package com.michitai.multiplayer.rooms.updates;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.ApiResponse;

import java.util.List;

/**
 * Response returned when updates are successfully sent to target players.
 */
public class UpdatePlayersResponse extends ApiResponse {
    @JsonProperty("updates_sent")
    private int updatesSent;

    @JsonProperty("update_ids")
    private List<String> updateIds;

    @JsonProperty("target_players_ids")
    private List<Integer> targetPlayersIds;

    public int getUpdatesSent() {
        return updatesSent;
    }

    public void setUpdatesSent(int updatesSent) {
        this.updatesSent = updatesSent;
    }

    public List<String> getUpdateIds() {
        return updateIds;
    }

    public void setUpdateIds(List<String> updateIds) {
        this.updateIds = updateIds;
    }

    public List<Integer> getTargetPlayersIds() {
        return targetPlayersIds;
    }

    public void setTargetPlayersIds(List<Integer> targetPlayersIds) {
        this.targetPlayersIds = targetPlayersIds;
    }
}
